hyperspy/_components/pes_voigt.py

```diff
@@ -0,0 +1,331 @@
+
+
+sqrt2pi = math.sqrt(2 * math.pi)
+sigma2fwhm = 2 * math.sqrt(2 * math.log(2))
+
+
+def voigt(x, FWHM=1, gamma=1, center=0, scale=1):
+    r"""Voigt lineshape.
+
+    The voigt peak is the convolution of a Lorentz peak with a Gaussian peak:
+    
+    .. math:: 
+        f(x) = G(x) \cdot L(x) 
+
+    where :math:`G(x)` is the Gaussian function and :math:`L(x)` is the 
+    Lorentzian function. In this case using an approximate formula by David 
+    (see Notes). This approximation improves on the pseudo-Voigt function
+    (linear combination instead of convolution of the distributions) and is,
+    to a very good approximation, equivalent to a Voigt function:
+
+    .. math:: 
+        z(x) &= \frac{x + i \gamma}{\sqrt{2} \sigma} \\
+        w(z) &= \frac{e^{-z^2} \text{erfc}(-i z)}{\sqrt{2 \pi} \sigma} \\
+        f(x) &= A \cdot \Re\left\{ w \left[ z(x - x_0) \right] \right\}
+
+
+    ============== =============
+    Variable        Parameter 
+    ============== =============
+    :math:`x_0`     center 
+    :math:`A`       scale
+    :math:`\gamma`  gamma
+    :math:`\sigma`  sigma
+    ============== =============
+    
+
+    Parameters
+    ----------
+    gamma : real
+       The half-width half-maximum of the Lorentzian.
+    FWHM : real
+       The FWHM=:math:`2 \sigma \sqrt(2 \log(2))` of the Gaussian.
+    center : real
+       Location of the center of the peak.
+    scale : real
+       Value at the highest point of the peak.
+
+    Notes
+    -----
+    Ref: W.I.F. David, J. Appl. Cryst. (1986). 19, 63-64
+    doi:10.1107/S0021889886089999
+    """
+    # wofz function = w(z) = Fad[d][e][y]eva function = exp(-z**2)erfc(-iz)
+    from scipy.special import wofz
+    sigma = FWHM / 2.3548200450309493
+    z = (np.asarray(x) - center + 1j * gamma) / (sigma * math.sqrt(2))
+    V = wofz(z) / (math.sqrt(2 * np.pi) * sigma)
+    return scale * V.real
+
+
+class Voigt(Component):
+    # Legacy class to be removed in v2.0
+
+    """This is the legacy Voigt profile component dedicated to photoemission 
+    spectroscopy data analysis that will renamed to `PESVoigt` in v2.0. To use
+    the new Voigt lineshape component set `legacy=False`. See the
+    documentation of ``hyperspy._components.voigt.Voigt`` for details on the
+    usage of the new Voigt component. 
+
+    .. math:: 
+        f(x) = G(x) \cdot L(x) 
+
+    where :math:`G(x)` is the Gaussian function and :math:`L(x)` is the 
+    Lorentzian function. This component uses an approximate formula by David 
+    (see Notes).
+
+    Attributes
+    ----------
+
+    legacy : Bool
+    area : Parameter
+    centre: Parameter
+    FWHM : Parameter
+    gamma : Parameter
+    resolution : Parameter
+    shirley_background : Parameter
+    non_isochromaticity : Parameter
+    transmission_function : Parameter
+    spin_orbit_splitting : Bool
+    spin_orbit_branching_ratio : float
+    spin_orbit_splitting_energy : float
```

Comment:
I realise that this is due to become legacy, but shouldn't the attributes that are specific to PESVoigt not be in this docstring?

Reply:
OK, I'll replace it by a note that refers to the two respective components. Anyway, it is called `Parameters` in other components ;-)